app/components/AppletData/index.js

```diff
@@ -131,19 +131,40 @@ class AppletData extends React.Component {
               const currentContraint = currentItem.valueConstraints;
               const oldConstraint = oldItem.valueConstraints;
 
-              currentContraint.minValue = min(oldConstraint.minValue, currentContraint.minValue);
-              currentContraint.maxValue = max(oldConstraint.maxValue, currentContraint.maxValue);
-
               const lang = currentContraint.itemList && Object.keys(currentContraint.itemList[0].name)[0]
-                        || oldConstraint.itemList && Object.keys(oldConstraint.itemList[0].name)[0];
+                        || oldConstraint.itemList && Object.keys(oldConstraint.itemList[0].name)[0]
+                        || 'en';
+
+              /** merge two sliders */
+              let currentRange = [0, 0];
+              let oldRange = [0, 0];
+
+              if (currentContraint.itemList.length) {
+                currentRange = [
+                  currentContraint.itemList[0].value, 
+                  currentContraint.itemList[currentContraint.itemList.length-1].value + 1
+                ];
+              }
+
+              if (oldConstraint.itemList.length) {
+                oldRange = [
+                  oldConstraint.itemList[0].value, 
+                  oldConstraint.itemList[oldConstraint.itemList.length-1].value + 1
+                ];
+              }
+
+              const range = [
+                currentRange[0] < oldRange[0] ? currentRange[0] : oldRange[0],
```

Comment:
This works, but it would be easier to read `Math.min(currentRange[0], oldRange[0])`